tests/Tokenizer/Analyzer/AttributeAnalyzerTest.php

```diff
@@ -129,4 +130,286 @@ public static function provideIsAttributeCases(): iterable
 
         yield [true, '<?php #[Foo("(")] class Bar {}'];
     }
+
+    /**
+     * @requires PHP 8.0
+     *
+     * @dataProvider provideGetAttributeDeclarationsCases
+     *
+     * @param array{0: string, 1: list<AttributeAnalysis>} $expected
+     */
+    public function testGetAttributeDeclarations(string $code, int $startIndex, array $expected): void
+    {
+        $tokens = Tokens::fromCode($code);
+        $attributeDeclarationAnalyses = AttributeAnalyzer::collect($tokens, $startIndex);
+
+        self::assertSame(
+            serialize($expected),
+            serialize($attributeDeclarationAnalyses),
+        );
+    }
+
+    /**
+     * @return iterable<array{0: string, 1: int, 2: list<AttributeAnalysis>}>
+     */
+    public static function provideGetAttributeDeclarationsCases(): iterable
+    {
+        yield [
+            '<?php
+            /**
+             * Start docblock
```

Comment:
Yes, this is intentional. There's really no way to know whether a certain comment or docblock belongs to the attribute or to the entire method. Since it's more common to add a docblock for a method at the beginning, before any attributes, while adding docblocks or comments between attributes isn't, this approach made the most sense to me.